src/Compilers/CSharp/Test/Semantic/Semantics/LookupPositionTests.cs

```diff
@@ -2043,6 +2043,81 @@ interface C : Base(X)
 
+        [Fact]
+        public void RecordInitializers_01()
+        {
+            var text = @"
+record C(int X)
+`{
+    int Z `= X + 1`;
+`}
+";
+            var expectedNames = MakeExpectedSymbols(
+                Add( // Global
+                    "System",
+                    "Microsoft",
+                    "C"),
+                Add( // Members
+                    "System.Boolean C.Equals(C? )",
+                    "System.Boolean C.Equals(System.Object? )",
+                    "System.Boolean System.Object.Equals(System.Object obj)",
+                    "System.Boolean System.Object.Equals(System.Object objA, System.Object objB)",
+                    "System.Boolean System.Object.ReferenceEquals(System.Object objA, System.Object objB)",
+                    "System.Int32 C.GetHashCode()",
+                    "System.Int32 C.X { get; init; }",
+                    "System.Int32 C.Z",
+                    "System.Int32 System.Object.GetHashCode()",
+                    "System.Object System.Object.MemberwiseClone()",
+                    "System.String System.Object.ToString()",
+                    "System.Type C.EqualityContract { get; }",
+                    "System.Type System.Object.GetType()",
+                    "void System.Object.Finalize()"),
+                Combine(
+                    Remove("System.Int32 C.X { get; init; }"),
+                    Add("System.Int32 X")
+                    ),
+                Combine(s_pop, s_pop),
+                s_pop
+            );
+
+            TestLookupNames(text, expectedNames);
+        }
+
+        [Fact]
+        public void RecordInitializers_02()
+        {
+            var text = @"
+record C(int X)
+`{
+    static int Z = X + 1;
+`}
+";
+            var expectedNames = MakeExpectedSymbols(
+                Add( // Global
+                    "System",
+                    "Microsoft",
+                    "C"),
+                Add( // Members
+                    "System.Boolean C.Equals(C? )",
+                    "System.Boolean C.Equals(System.Object? )",
+                    "System.Boolean System.Object.Equals(System.Object obj)",
+                    "System.Boolean System.Object.Equals(System.Object objA, System.Object objB)",
+                    "System.Boolean System.Object.ReferenceEquals(System.Object objA, System.Object objB)",
+                    "System.Int32 C.GetHashCode()",
+                    "System.Int32 C.X { get; init; }",
+                    "System.Int32 C.Z",
+                    "System.Int32 System.Object.GetHashCode()",
+                    "System.Object System.Object.MemberwiseClone()",
+                    "System.String System.Object.ToString()",
+                    "System.Type C.EqualityContract { get; }",
+                    "System.Type System.Object.GetType()",
+                    "void System.Object.Finalize()"),
+                s_pop
+            );
+
+            TestLookupNames(text, expectedNames);
+        }
```

Comment:
Consider testing a `const` field as well.

Reply:
> Consider testing a const field as well.

Const fields are static fields, so I am pretty confident this test is fairly equivalent to verify scoping rules. I will add a test like that for completeness sake in a separate PR 

---
In reply to: [436281327](https://github.com/dotnet/roslyn/pull/44906#discussion_r436281327) [](ancestors = 436281327)